libs/ha/ha_common_libs_k8s.py

```diff
@@ -1480,3 +1483,109 @@ def delete_single_pod_setting_replica_0(self, master_node_obj, health_obj,
             return False, resp
         LOGGER.info("Services of pod are in online state")
         return True, pod_name
+
+    def mark_resource_failure(self, mnode_obj, pod_list: list, go_random: bool = False,
+                              rsc_opt: str = "mark_node_failure", rsc: str = "node",
+                              validate_set: bool = True):
+        """
+        Helper function to set resource status to Failed random if go_random.
+        :param pod_list: List of resource to be marked as failed
+        :param go_random: If True, send mark failure signal to resource randomly
+        :param mnode_obj: Master node object to fetch the resource ID
+        :param rsc_opt: Operation to be performed on resource (eg. mark_node_failure)
+        :param rsc: resource type (eg. node, cluster)
+        :param validate_set: If set to TRUE, its validate SET failure for resource
+        :return: bool, response
+        """
+        if rsc == 'node':
+            pod_info = {}
+            pod_data = {'id': None, 'status': 'offline'}
+            for pod in pod_list:
+                pod_info[pod] = pod_data.copy()
+                pod_info[pod]['id'] = mnode_obj.get_machine_id_for_pod(pod)
+
+            for pod in pod_list:
+                set_fail = self.system_random.choice([True, False]) if go_random else True
+                if set_fail:
+                    LOGGER.info('Marking %s pod as failed', pod)
+                    data_val = {"operation": rsc_opt,
+                                "arguments": {"id": f"{pod_info[pod]['id']}"}}
+                    resp = self.system_health.set_resource_signal(req_body=data_val)
+                    if not resp[0]:
+                        return False, pod_info, f"Failed to set failure status for {pod}"
+                    pod_info[pod]['status'] = 'failed'
+            if validate_set:
+                return self.get_validate_resource_status(rsc_info=pod_info)
+            return True, pod_info, f"{pod_list} marked as failed"
+        return False, f"Mark failure for {rsc} is not supported yet"
+
+    def get_validate_resource_status(self, rsc_info=None, exp_sts: str = "failed",
+                                     mnode_obj=None, rsc: str = "node"):
+        """
+        Helper function to get and validate resource status
+        :param exp_sts: Expected status of resource.
+        :param rsc_info: Required resource to get its status
+        :param rsc: resource type (eg. node, cluster)
+        :param mnode_obj: Master node object to fetch the resource ID
+        :return: bool, response
+        """
+        if rsc == "node":
+            if not isinstance(rsc_info, dict):
+                # Get the node ID and set expected status if only list of pods is passed
+                pod_info = {}
+                pod_data = {'id': None, 'status': "offline"}
+                for pod in rsc_info:
```

Comment:
Do you expect rsc_info to be list here? If yes, please mention the same in doc string as well.